File: Sources/WebApp/src/app/modules/core/utils/patterns.ts
import {FormControl} from '@angular/forms';

export abstract class Patterns {
  static EMAIL_PATTERN = '^([a-zA-Z0-9_\\-\\.]+)@([a-zA-Z0-9_\\-\\.]+)\\.([a-zA-Z]{2,5})$';
  static PASSWORD_PATTERN = '^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*\\W+).{9,15}$';
  static PHONE_PATTERN = '^\\+[0-9]+$';
  static PHONE_CODE_PATTERN = '^[0-9]{6}$';
  static DOMAINS_PATTERN = `sytoss.com;mexem.com`;
  // illegalDomain error
  static ACCOUNT_NAME_PATTERN = '^[A-Za-z]{1}\\d{6,7}$';
  static checkDomain(f: FormControl): any {
    const domains = Patterns.DOMAINS_PATTERN.split(';');
    return !!domains.find(v => f.value.includes(v)) ? null : { illegalDomain: true };
  }
}

File: Sources/WebApi/BusinessLayer/IBAR.TradeModel.Business/Common/Patterns.cs
namespace IBAR.ServiceLayer.Common
{
    public static class Patterns
    {
        public const string EmailPattern = @"^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$";
        public const string PasswordPattern = @"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*\W+).{9,15}$";
        public const string PhonePattern = @"^\+[0-9]+$";
        public const string NamePattern = @"^[a-zA-Z]+$";
        public const string AccountName = @"^[A-Za-z]{1}\d{5,7}$";
        public const string PhoneCodePattern = @"^\d{6}$";
    }
}